cluster/src/main/java/org/apache/iotdb/cluster/entity/raft/DataStateMachine.java

```diff
@@ -82,59 +76,77 @@ public DataStateMachine(String groupId, PeerId peerId) {
   @Override
   public void onApply(Iterator iterator) {
     while (iterator.hasNext()) {
-
-      Closure closure = null;
-      DataGroupNonQueryRequest request = null;
-      BasicResponse response = null;
+      final Closure closure = iterator.done();
       final ByteBuffer data = iterator.getData();
+
+//      /** It's leader to apply task **/
+//      if (closure != null) {
+//        RaftTaskManager.getInstance().execute(() -> applySingleTask(closure, data));
+//      } else {
+        applySingleTask(closure, data);
+//      }
+      iterator.next();
+    }
+  }
+
+  /**
+   * Apply a single raft task. If this node is the leader of state machine's data group, apply the
+   * task in thread.
+   *
+   * @param closure if this node is leader, closure is not null.
+   * @param data Request data
+   */
+  private void applySingleTask(Closure closure, ByteBuffer data) {
+    BasicResponse response = (closure == null) ? null : ((ResponseClosure) closure).getResponse();
+    DataGroupNonQueryRequest request = null;
+    try {
+      request = SerializerManager.getSerializer(SerializerManager.Hessian2)
+          .deserialize(data.array(), DataGroupNonQueryRequest.class.getName());
+    } catch (final CodecException e) {
+      LOGGER.error("Fail to decode IncrementAndGetRequest", e);
+    }
+
+    Status status = Status.OK();
+    assert request != null;
+
+    List<byte[]> planBytes = request.getPhysicalPlanBytes();
+
+    LOGGER.debug(String.format("State machine batch size(): %d", planBytes.size()));
```

Comment:
LOGGER.debug("State machine batch size(): %d", planBytes.size());

Reply:
It will be a sonar issue with your way.